edx_sysadmin/views.py

```diff
@@ -34,44 +33,88 @@
     get_course_by_id,
     get_registration_required_extra_fields_with_values,
     is_registration_api_functional,
+    user_has_access_to_users_panel,
+    user_has_access_to_courses_panel,
+    user_has_access_to_git_logs_panel,
+    user_has_access_to_git_import_panel,
 )
+from edx_sysadmin.utils.decorators import user_has_sysadmin_access
+
 
 log = logging.getLogger(__name__)
 
 
+@method_decorator(user_has_sysadmin_access, name="dispatch")
+class SysadminDashboardRedirectionView(RedirectView):
+    """Redirection view to land user to specific panel"""
+
+    def get_redirect_url(self, *args, **kwargs):
+        """Override redirection_url"""
+
+        if user_has_access_to_users_panel(self.request.user):
+            return reverse("sysadmin:users")
+        elif user_has_access_to_courses_panel(self.request.user):
+            return reverse("sysadmin:courses")
+        elif user_has_access_to_git_logs_panel(self.request.user):
+            return reverse("sysadmin:gitlogs")
+        elif user_has_access_to_git_import_panel(self.request.user):
+            return reverse("sysadmin:gitimport")
+        else:
+            raise Http404
+
+
 @method_decorator(ensure_csrf_cookie, name="dispatch")
-@method_decorator(staff_member_required, name="dispatch")
+@method_decorator(user_has_sysadmin_access, name="dispatch")
```

Comment:
We should use `user_passes_test` decorator here and pass that decorator our custom validation method `user_is_staff` or `user_has_course_access` and put the logic of required permissions in those methods.

Reply:
I have updated the permission decorators.